### DISCUSSION TITLE
How to obtain separate region masks for region-based training?

### DISCUSSION THREAD

**Discussion by Dylan-H-Wang** [MIC-DKFZ/nnUNet#1439] [Q&A]
Hi,

I am using nnUNet to train on a dataset which have overlapped labels similar to BraTS 2020. However, there are some differences in my case that the single pixel can belong to multiple classes, while in BraTS 2020, each pixel only belongs to one class. Therefore, the generated mask (prediction results) cannot solve my problem.

Here is how I define regions:
```python
origin_mask={
    "background": 0,
    "cell_membrane": 1,
    "nano": 2,
    "nano_cell": 3 #pixels which are overlaps of nano and cell membrane
}

# in dataset.json
labels={
    "background": 0,
    "cell_membrane": [1, 3],
    "nano": [2, 3],
}
regions_class_order=(1, 2)
```

By training nnUNet using above region definition, the generated mask (inference results) will treat the overlap of cell membrane and nano as nano only, which should also belong to cell membrane.

One solution I can think of is to add another region definition `"nano_cell": [3]`. However, I am concerning this may affect the training of model since predicting the overlap parts sounds a bit of trivial.

Is there any way to get separate region masks for each of defined region by `nnUNetv2_predict`?

Thank you!

**Comment by KanielDatz** [comment]
I  have a similar problem; a catheter, vessel, and their overlap.
have you managed to solve this?
